orderer/common/cluster/block_fetcher.go

```diff
@@ -343,19 +362,20 @@ func (bf *BlockFetcher) maybeUpdateLatestConfigBlock(block *common.Block) {
 	}
 }
 
-func (bf *BlockFetcher) shuffleEndpoint() {
+func (bf *BlockFetcher) shuffleEndpoint() error {
 	candidates := bf.blockSourceCandidates()
 	// handle case when candidates list is empty
 	if len(candidates) == 0 {
 		bf.Logger.Info("Can't shuffle blockpuller endpoint. Not enough endpoints available")
-		return
+		return errors.New("can't shuffle blockpuller endpoint. Not enough endpoints available")
 	}
 	bf.shuffleIndex++
 	effectiveIndex := bf.shuffleIndex % len(candidates)
 	bf.currentEndpoint = candidates[effectiveIndex]
 	bf.Logger.Debugf("Shuffled block puller endpoint. New endpoint: [%s]", bf.currentEndpoint)
-	bf.setBlockSource(bf.currentEndpoint)
+	err := bf.setBlockSource(bf.currentEndpoint)
 	bf.lastShuffledAt = bf.TimeNow()
```

Comment:
if you failed shuffling, why anyway set the time of shuffle if you didn't do that?